assets/src/models/createDetourMachine.ts

```diff
@@ -1,23 +1,114 @@
 
 export const createDetourMachine = setup({
   types: {} as {
-    events: { type: "detour.share" } | { type: "detour.resume-editing" }
+    context: {
+      waypoints: ShapePoint[]
+    }
+    events:
+      | { type: "detour.share" }
+      | { type: "detour.resume-editing" }
+      | { type: "detour.edit.place-waypoint"; location: ShapePoint }
+      | { type: "detour.edit.place-connection-point"; location: ShapePoint }
+      | { type: "detour.edit.clear-detour" }
+      | { type: "detour.edit.undo" }
+      | { type: "detour.edit.done" }
+      | { type: "detour.share.copy-detour"; detourText: string }
+  },
+  actions: {
+    "detour.add-waypoint": assign({
+      waypoints: ({ context }, params: { location: ShapePoint }) => [
+        ...context.waypoints,
+        params.location,
+      ],
+    }),
+    "detour.clear": assign({
+      waypoints: [],
+    }),
+    "detour.undo": assign({
+      waypoints: ({ context }) => context.waypoints.slice(0, -1),
+    }),
+    "detour.share.copy-detour": (_, { detourText }: { detourText: string }) => {
+      window.navigator.clipboard?.writeText(detourText)
+    },
   },
 }).createMachine({
   id: "Detours Machine",
+  context: () => ({
+    waypoints: [],
+  }),
 
   initial: "Detour Drawing",
   states: {
     "Detour Drawing": {
       initial: "Editing",
       states: {
+        // ? Can this be modeled with a sub state machine as an actor?..
         Editing: {
-          initial: "Idle",
+          initial: "Pick Start Point",
+          on: {
+            "detour.edit.clear-detour": {
+              target: "Editing",
+              actions: "detour.clear",
+            },
+          },
           states: {
-            Idle: {
+            "Pick Start Point": {
+              on: {
+                "detour.edit.place-connection-point": {
+                  target: "Place Waypoint",
+                  actions: [
+                    {
+                      type: "detour.add-waypoint",
+                      params: ({ event }) => ({
+                        location: event.location,
+                      }),
+                    },
+                  ],
+                },
+              },
+            },
+            "Place Waypoint": {
               on: {
-                "detour.share": {
+                "detour.edit.place-waypoint": {
+                  actions: [
+                    {
+                      type: "detour.add-waypoint",
+                      params: ({ event }) => ({
+                        location: event.location,
+                      }),
+                    },
+                  ],
+                },
+                "detour.edit.place-connection-point": {
+                  target: "Finished Drawing",
+                  actions: [
+                    {
+                      type: "detour.add-waypoint",
+                      params: ({ event: { location } }) => ({
+                        location,
+                      }),
```

Comment:
Ok for some reason, the latest pushes made gh lose track of this comment, so I'm going to find another example and attach a new comment (since I see it's still there)

Reply:
cont. https://github.com/mbta/skate/pull/2607#discussion_r1611805683